datafusion/core/src/datasource/listing_table_factory.rs

```diff
@@ -59,33 +57,22 @@ impl TableProviderFactory for ListingTableFactory {
         state: &SessionState,
         cmd: &CreateExternalTable,
     ) -> datafusion_common::Result<Arc<dyn TableProvider>> {
-        let file_compression_type = FileCompressionType::from(cmd.file_compression_type);
+        let table_config = create_table_options_from_cmd(state.config_options(), &cmd)?;
```

Comment:
We created the table config here, now formats can use this to fetch their options.